docs/preview/features/secret-store/named-secret-providers.md

```diff
@@ -0,0 +1,46 @@
+---
+title: "Named secret providers"
+layout: default
+---
+
+# Retrieve a specific secret provider from the secret store
+
+The default workings of the secret store, is that a set of secret providers are registered and the consumer gets access to all of secrets provided by `ISecretProvider`s.
+In some cases, you may want to retrieve a specific secret provider from the store. This can be because that secret provider has some functionality that the other providers doesn't have.
+
+In those cases, you can register your secret provider with a unique name. This name will be matched with the secret provider so that in a later stage, you can retrieve back your named provider.
+
+## Register a named secret provider
+
+Lets consider that you want explicitly use the built-in environment variables secret provider.
+First, the secret provider has to be registered with a unique name. 
+
+```csharp
+.ConfigureSecretStore((config, stores) =>
+{
+    stores.AddEnvironmentVariables(..., name: "environment-variables");
+})
+```
+
```

Comment:
```suggestion

⚠ The name of the registered secret providers should be unique; otherwise, an exception will be thrown when you try to access the `GetProvider` or `GetCachedProvider`.

```

Reply:
However, for the last part I'd argue that the secret store builder should verify and block it so that it fails at startup, not runtime. Do you agree?

Reply:
Yes, that would be better. I'll make sure we do that. Although, we can only verify it for the current secret store setup, not across setups. Let me clarify, we can't still guard against this:

```csharp
.ConfigureSecretStore(stores => stores.AddProvider(..., name: "duplicate-name"))
.ConfigureSecretStore(stores => stores.AddProvider(..., name: "duplicate-name"))
```